strawberryfields/ops.py

```diff
@@ -1852,31 +1839,43 @@ def _decompose(self, reg):
 
         if self.pure and is_diag:
             # covariance matrix consists of x/p quadrature squeezed state
-            for n, expr in enumerate(D[:self.ns]*2/self.hbar):
-                if np.abs(expr - 1) >= _decomposition_merge_tol:
+            for n, expr in enumerate(D[:self.ns]):
+                if np.abs(expr - 1) >= _decomposition_tol:
                     r = abs(log(expr)/2)
-                    cmds.append(Command(Sgate(r, 0), reg[n]))
+                    cmds.append(Command(Squeezed(r, 0), reg[n]))
+                else:
+                    cmds.append(Command(Vac, reg[n]))
 
         elif self.pure and is_block_diag:
             # covariance matrix consists of rotated squeezed states
             for n, v in enumerate(BD_modes):
-                if not np.all(v - np.identity(2)*self.hbar/2 < 1e-10):
-                    r = np.abs(arccosh(np.sum(np.diag(v/self.hbar)))/2)
-                    phi = arctan(2*v[0, 1] / np.sum(np.diag(v)*[1, -1]))
-                    cmds.append(Command(Sgate(r, phi), reg[n]))
+                if not np.all(v - np.identity(2) < 1e-10):   # FIXME magic tolerance
```

Comment:
Who wrote this code? We should ask them the purpose of this tolerance. Was it @nquesada?

Reply:
There are lots of tolerance checks in the Decomposition code, and some of them might be hard to justify exactly. I just noticed that this one used a numerical value instead of _decomposition_tol.

Reply:
Can we replace it with `_decomposition_tol` for consistency?

Reply:
That was my thought as well

Reply:
Ok